.github/workflows/ci_build.yml

```diff
@@ -21,24 +22,33 @@ jobs:
         with:
           python-version: ${{ matrix.python-version }}
 
-      - name: Install dependencies
+      - name: Caching dependencies
+        uses: actions/cache@v2
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-py-${{ matrix.python-version }}-pytorch-${{ matrix.pytorch-version }}-pip-${{ hashFiles('**/setup.py') }}
+          restore-keys: |
+            ${{ runner.os }}-py-${{ matrix.python-version }}-pytorch-${{ matrix.pytorch-version }}-pip-
+
+      - name: Install dependencies with PyTorch ${{ matrix.pytorch-version }}
+        if: steps.cache.outputs.cache-hit != 'true'
         run: |
           python -m pip install --upgrade pip
+          pip install torch==${{ matrix.pytorch-version }}
           pip install -e .[tests]
+
       - name: Lint with flake8
         run: |
           pip install flake8
           # stop the build if there are Python syntax errors or undefined names
           flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
-          # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
-          flake8 . --count --exit-zero --max-complexity=15 --max-line-length=127 --statistics
 
       - name: Run tests and generate coverage
         run: |
           pytest --cov=torch_lr_finder --cov-report=xml tests/
-          
+
       - name: Upload coverage to Codecov
-        if: matrix.python-version == 3.8
+        if: matrix.python-version == 3.8 && matrix.pytorch-version == '1.5.0'
```

Comment:
This line can be removed to make code coverage report of each job be uploaded to codecov.